### PR TITLE
mysql/decimal: fix bug in decode

### PR DESCRIPTION
**Pull request by AndreMouche**
fix issue https://github.com/pingcap/tidb/issues/5271  @BusyJay @coocood PTAL

### CODE REVIEW

File: src/coprocessor/codec/mysql/decimal.rs [hunk 1]
```diff
@@ -1941,7 +1941,7 @@ pub trait DecimalDecoder: BytesDecoder {
             if d.word_buf[word_idx] >= TEN_POW[leading_digits + 1] {
                 return Err(box_err!("invalid leading digits for decimal number"));
             }
-            if d.word_buf[word_idx] != 0 {
+            if word_idx > 0 || d.word_buf[word_idx] != 0 {
```
Comment (BusyJay) [inline src/coprocessor/codec/mysql/decimal.rs]: When will `word_idx > 0`?

Comment (coocood) [reply src/coprocessor/codec/mysql/decimal.rs]: This expression is always false, but it is more independent, it doesn't depend on the fact that word_idx is initialized to 0.

Comment (BusyJay) [reply src/coprocessor/codec/mysql/decimal.rs]: What can be benefit from this? Can the function still work if `word_idx` is initialized to non-zero?

**Comment by coocood** [comment]
LGTM

**Pull request merged**